src/cmd/graphql.go

```diff
@@ -89,138 +92,138 @@ query ($id: ID!){
-            handleErr("error making graphql api call", err)
-            for _, item := range handleAggregate(data, aggregation) {
-                output = append(output, item)
-            }
-
-            if paginate {
-                hasNextPage, err = strconv.ParseBool(string(hasNextPageExp.FindSubmatch(data)[1]))
-                handleErr("error parsing bool for has next page", err)
-                variables["endCursor"] = string(endCursorExp.FindSubmatch(data)[1])
-            } else {
-                hasNextPage = false
-            }
-        }
-
-        json, err := json.Marshal(output)
-        handleErr("error marshaling output to json", err)
-
-        fmt.Println(string(json))
-    },
+	Run: func(cmd *cobra.Command, args []string) {
+		flags := cmd.Flags()
+		headersValue, err := flags.GetStringArray("header")
+		headers := map[string]string{}
+		for _, value := range headersValue {
+			matches := keyValueExp.FindStringSubmatch(value)
+			headers[matches[1]] = matches[2]
+		}
+
+		cobra.CheckErr(err)
+		paginate, err := flags.GetBool("paginate")
+		handleErr("error getting paginate flag", err)
+
+		aggregate, err := flags.GetString("aggregate")
+		handleErr("error getting aggregate flag", err)
+		jq, err := gojq.Parse(aggregate)
+		handleErr("error parsing pagination flag value", err)
+		aggregation, err := gojq.Compile(jq)
+
+		handleErr("error compiling pagination flag value", err)
+		queryValue, err := flags.GetString("query")
+		handleErr("error getting query flag value", err)
+		queryParsed, err := convert(queryValue)
+		handleErr("error parsing query flag value", err)
+		query, ok := queryParsed.(string)
+		if !ok {
+			handleErr("error parsing query flag value", fmt.Errorf("'%#v' is not a string", queryParsed))
+		}
+		operationName, err := flags.GetString("operationName")
+		cobra.CheckErr(err)
+		fields, err := flags.GetStringArray("field")
+		handleErr("error getting field flag value", err)
+
+		variables := map[string]interface{}{}
+		for _, field := range fields {
+			matches := keyValueExp.FindStringSubmatch(field)
+			value, err := convert(matches[2])
+			handleErr(fmt.Sprintf("error parsing variable '%s'", field), err)
+			variables[matches[1]] = value
+		}
+
+		client := getClientGQL(opslevel.SetHeaders(headers))
+		var output []interface{}
+
+		hasNextPage := true
+		for hasNextPage {
+			data, err := client.ExecRaw(query, variables, opslevel.WithName(operationName))
+			handleErr("error making graphql api call", err)
+			output = append(output, handleAggregate(data, aggregation)...)
+
+			if paginate {
+				hasNextPage, err = strconv.ParseBool(string(hasNextPageExp.FindSubmatch(data)[1]))
+				handleErr("error parsing bool for has next page", err)
+				variables["endCursor"] = string(endCursorExp.FindSubmatch(data)[1])
+			} else {
+				hasNextPage = false
+			}
+		}
+
+		json, err := json.Marshal(output)
+		handleErr("error marshaling output to json", err)
+
+		fmt.Println(string(json))
+	},
```

Comment:
Where there any changes in here besides spaces?  Its unclear to me.

Reply:
I think there was one place that changed a for loop to an append(). But this is a mess. These changes will be introduced in a follow up PR